llvm-project/stage1.cmake

```diff
@@ -1,45 +1,134 @@
-# TODO: comment every one of these.
+# Please try to keep these cmake variables alphabetically sorted.
+
+# Enable optimizations, as opposed to a debug build.
 set(CMAKE_BUILD_TYPE "Release" CACHE STRING "")
+
+# Use Alpine's clang and clang++ from their clang package as the stage0
+# compilers.
 set(CMAKE_CXX_COMPILER "/usr/bin/clang++" CACHE FILEPATH "")
 set(CMAKE_C_COMPILER "/usr/bin/clang" CACHE FILEPATH "")
+
+# Alpine's clang does not default to LLD; it defaults to BFD. Use LLD when
+# building executables and shared libraries.
 set(CMAKE_EXE_LINKER_FLAGS "-fuse-ld=lld" CACHE STRING "")
 set(CMAKE_SHARED_LINKER_FLAGS "-fuse-ld=lld" CACHE STRING "")
+
+# Set the default target triple to match the host.
+# TODO: passing in the value of $(clang -print-multiarch) causes failures.
+# It seems that alpine clang's default target triple is x86_64-linux-gnu.
+# Perhaps missing alpine in the triple causes some incompatibility?
 set(LLVM_DEFAULT_TARGET_TRIPLE "x86_64-alpine-linux-musl" CACHE STRING "")
+
+# Use Alpine's lld as the stage0 linker to link everything.
 set(LLVM_ENABLE_LLD ON CACHE BOOL "")
+
+# For stage 1, just build clang and LLD, which are used by the "runtimes build"
+# to build the runtime libraries of LLVM.
 set(LLVM_ENABLE_PROJECTS "clang;lld;" CACHE STRING "")
+
+# LLVM "runtimes" build.
 set(LLVM_ENABLE_RUNTIMES "compiler-rt;libcxx;libcxxabi;libunwind;" CACHE STRING "")
+
+# FORCE_ON causes the build to fail if zlib is not found in the environment
+# during configuration, rather than much later during link.
 set(LLVM_ENABLE_ZLIB "FORCE_ON" CACHE STRING "")
+
+# Just build stage1 to target the host. It's not the end product, so it won't
+# be able to target all of the kernel targets we can build.
 set(LLVM_TARGETS_TO_BUILD "host;" CACHE STRING "")
+
+# Stage1 clang should default to lld rather than BFD.
+# TODO: it seems this was ignored by the runtimes build...
+# otherwise why would CMAKE_EXE_LINKER_FLAGS be necessary?
```

Comment:
This embeds the default value for the new driver, it doesn't control the linking itself.

Reply:
yep, I wrote the stage1 comments yesterday, but the stage2 comments (which are more cohesive) this morning.  Probably further case for @nathanchance 's recommendation to include a common base; then the comments don't need to stay in sync since they only exist on one place.

Will fix.